driver-core/src/main/com/mongodb/internal/connection/ClientMetadataHelper.java

```diff
@@ -104,45 +98,17 @@ private static boolean nameMatches(final String name, final String... prefixes)
         return false;
     }
 
-    private static String getDriverVersion() {
-        String driverVersion = "unknown";
-        String path = getCodeSourcePath();
-        if (path != null) {
-            try {
-                URL jarUrl = path.endsWith(".jar") ? new URL("jar:file:" + path + "!/") : null;
-                if (jarUrl != null) {
-                    JarURLConnection jarURLConnection = (JarURLConnection) jarUrl.openConnection();
-                    Manifest manifest = jarURLConnection.getManifest();
-                    if (manifest != null) {
-                        Attributes mainAttributes = manifest.getMainAttributes();
-                        if (mainAttributes != null) {
-                            String version = (String) mainAttributes.get(new Attributes.Name("Build-Version"));
-                            if (version != null) {
-                                driverVersion = version;
-                            }
-                        }
-                    }
-                }
-            } catch (IOException e) {
-                // do nothing
-            }
-        }
-        return driverVersion;
-    }
-
-    private static String getCodeSourcePath() {
-        String path = null;
-        ProtectionDomain protectionDomain = InternalStreamConnectionInitializer.class.getProtectionDomain();
-        if (protectionDomain != null) {
-            CodeSource codeSource = protectionDomain.getCodeSource();
-            if (codeSource != null) {
-                URL location = codeSource.getLocation();
-                if (location != null) {
-                    path = location.getPath();
-                }
-            }
+    // The driver version is stored in a class generated by a build plugin, so it has to be accessed dynamically here
+    static String getDriverVersion() {
+        try {
+            return (String) Class.forName("com.mongodb.internal.connection.MongoDriverVersion").getDeclaredField("VERSION").get(null);
+        } catch (ClassNotFoundException e) {
```

Comment:
There's no good reason anymore why we should default to "unknown".  If this class doesn't exist, then something has gone wrong with the build process.

Reply:
You don't need to reflect at all just do:

```
import com.mongodb.internal.connection.MongoDriverVersion

static String getDriverVersion() {
    return MongoDriverVersion.VERSION;
}
```

It just requires the build step first otherwise intellij will complain, but I live with that in Scala without real issue.

Reply:
Also lets update `getDriverInformation` to get the driver name from `MongoDriverVersion` class also.

In the future we could potentially separate out `mongo-java-driver` and its variants (sync, async, legacy), which could be insightful.

Reply:
Even after building it doesn't compile in IntelliJ

Error:(115, 47) java: cannot find symbol
  symbol:   class MongoDriverVersion
  location: package com.mongodb.internal.connection

Are you able to compile without error on this branch?

Reply:
Can we wait to make this change to getDriverInformation until we want it to be something other than "mongo-java-driver"?  I don't see that it buys us anything now.

Reply:
I dont see a need for `mongo-java-driver` to be a string in the class, especially when we set it in the `MongoDriverVersion` we should just use it from there - if we can get intellij happy.